### DISCUSSION TITLE
What's unEuropean here?

### DISCUSSION THREAD

**Discussion by mcfnord** [jamulussoftware/jamulus#2047] [General]
Hi.

I wrote an app: https://jamulus.live

9 browsers worldwide are looking at it right now, but none are in Europe.

I have a theory that Western Europe is so active on Jamulus that finding the server you want is easy. In Germany, people often gather on servers right on the hour. Outside Europe, finding active servers gets more complicated, and additional tooling helps.

That's my theory. 

If you are in Europe, can you think of features you'd like to see?

John

**Comment by rluc99** [comment]
@mcfnord Great app! Really useful (I am based in UK). 
I like the way you have added times for people who are waiting. Also your refresh rate looks about right.
I'll have a think about extra features but I think you have it pretty well covered.

**Reply by gilgongo** [reply]
Assuming "right now" was nine hours ago, perhaps it was a bit late in Europe's time zones?  

BTW it's a bit hard to tell, but your list seems to only be a subset of the servers per directory, as compared to the full list here https://explorer.jamulus.io/

**Reply by pljones** [reply]
I just checked "Hide Empty Servers" on Jamulus Explorer and went through all the servers.  Apart from Jamulus Explorer showing Thomakaju - the dead one - all the servers match as far as I could see.

**Reply by gilgongo** [reply]
Ah, maybe I was fooled by an auto-refresh.

**Reply by rdica** [reply]
I am a user of @mcfnord site, and the fact that `Thomakaju` wasn't listed there even though its listed in the client explorer and website explorer was what prompted me to submit that issue. Thanks for this @mcfnord !

**Reply by rluc99** [reply]
@mcfnord I like your new GUI. However the ability to do a string search has disappeared. That was really useful if there are a lot of users on. Any plans to reinstate it?
Also, what determines the order of the display? It looks like it might be ordering by geography but then there are often exceptions.

**Reply by rdica** [reply]
@rluc99 Clicking on a username highlights it, and it's remembered. What I don't know is if that action is remembered through browser restarts. It is remembered if page/tab is closed and reopened.
You can click on multiple usernames if desired as well. When they come back on, they'll be highlighted.

**Reply by rluc99** [reply]
@rdica I see. Yes that works - pretty cool. 
The problem is though, that you can only find a name if you've clicked on it already. What if the name isn't there to click on?

**Reply by rdica** [reply]
If the user isn't on a server, a search didn't return anything. How is that different? You can always use the browsers text search if there's too many users to read through before clicking...

@mcfnord is open to ideas/suggestions, I've shared my thoughts and feedback with him occassionally on jamulus. Perhaps he'll reinstate a search function :)

**Reply by rluc99** [reply]
"You can always use the browsers text search" - now why didn't I think of that!! :)

**Reply by mcfnord** [reply]
@rluc99:

Hi.

The search bar has not returned but the Musicians list (down below) may help.

I don't list all occupied servers. If nobody moves for 6 hours, I make the server disappear, because I suspect it's actually silent, with just a parked user or two. I also hide servers that just contain a known probe. My goal is to only show people who are likely to be active, based on the clues I have.

I order servers by distance from the browser user as best I can. Servers with 2+ users are listed first, followed by servers with just one user. The geography list isn't always right because I respect the server's self-described location (Antartica?), and because getting geographic locale from IP address is somewhat unreliable. I use geography as a proxy for latency. Entering your own city into Jamulus and logging in increases the reliability of this feature.

Anyone can click a name, and that name-country match will appear in yellow. But users who log in get auto-tracking (in pink). There's no password, I just rely on your self-identification. I rely on these automated hints, which are based on duration I've spent with each musician on any public server, and then I click names of people who truly stand out. (Try ctrl-click for a secret feature!)

You have noticed that people who aren't on a public server do not appear on the tool. In this sense, all its features are a "live list" and I don't even track details of people who are offline. I track a hash but not a name.

The user count hovers around 20 now at the peak hour each day, and I believe the userbase is very loyal and consistent. I've localized key features to German, Thai, and Chinese.
